### PR TITLE
Update DG and UG

### PR DESCRIPTION
**Pull request by faqihisaninja**
Updated diagrams in the developer guide
Added implementation specific details of mark-attend command
Update User Guide to include mark-attend command details

### CODE REVIEW

**Review by yejiadong — APPROVED** [review]
LGTM.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -178,6 +178,49 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+The mark-attend mechanism extends `Command` with the ability to mark a student as having attended a particular training session.
+
+This feature makes use of the `Attend` class which each student keeps track of. As stated above, by default an `Attend` object is constructed with `hasAttended = false`. This command sets `hasAttended` to equal to `true`.
+
+The mark-attend command takes in a `Training` index and multiple `Student` ids as input. Any error in the input format will result in the whole command being discarded and the state of the canoe coach book will remain unchanged.
+A list of possible input errors are listed below:
+- `Training` index is out of range -> `Training` cannot be found
+- `Student` index is out of range -> `Student` cannot be found
+- Empty parameters. i.e. `Training` index and/or `Student` ids not input
+- `Student` does not have specified `Training` as per of it's schedule
```
Comment (yejiadong) [inline docs/DeveloperGuide.md]: small typo here - part of its schedule

**Review by yejiadong — COMMENTED** [review]
Small nits

**Review by Whleee — APPROVED** [review]
LGTM

**Pull request merged**